tree borrows: implicit writes (#4947)
rust-lang/miri: README.md

  track interior mutable data on the level of references instead of on the
  byte-level as is done by default.  Therefore, with this flag, Tree
  Borrows will be more permissive.

> * `-Zmiri-tree-borrows-implicit-writes` enables the tracking of implicit / spurious writes. This only has an effect in Miri if [Tree Borrows] is enabled. This makes Tree Borrows less permissive.

Reviewer: ```suggestion
* `-Zmiri-tree-borrows-implicit-writes` enables implicit writes for all `&mut` function arguments.
  This makes Tree Borrows less permissive.
```
Please also keep the list alphabetic.
